jetty-util/src/main/java/org/eclipse/jetty/util/security/CredentialFactory.java

```diff
@@ -0,0 +1,68 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2016 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.util.security;
+
+import java.util.ServiceLoader;
+
+import org.eclipse.jetty.util.security.Credential.Crypt;
+import org.eclipse.jetty.util.security.Credential.MD5;
+
+/**
+ * Class responsible for the creation of authentication credentials from strings.
+ * It supports both the known Credential types (CRYPT and MD5) and some 
+ * Credential types discovered at runtime by using the service loader mechanism.
+ *
+ * @see Credential
+ * @see Password
+ */
+public class CredentialFactory
+{
+	private ServiceLoader<CredentialProvider> credentialProviderLoader;
+
+	/**
+	 * Constructor
+	 */
+    public CredentialFactory() {
```

Comment:
curlies on new lines please. See the [coding standards](https://www.eclipse.org/jetty/documentation/current/coding-standards.html)